processor/simple_supply_tp/payload.py

```diff
@@ -32,6 +32,9 @@ def data(self):
         if self._transaction.HasField('create_agent'):
             return self._transaction.create_agent
 
+        if self._transaction.HasField('create_record'):
```

Comment:
Doing it in the way you described was because it would require disabling the [no-else-return](https://pylint.readthedocs.io/en/latest/technical_reference/features.html#id23) pylint message. 
Since they are functionally equivalent and the readability is not too much of an issue due to the simplicity of the method I didn't think it was worth it.